### ISSUE TITLE
Bad file descriptor error during cleanup of MirroredStrategy model

### ISSUE THREAD

**Issue by alexarmstrongvi** [tensorflow/tensorflow#56279]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.8.0

### Custom Code

No

### OS Platform and Distribution

Linux Red Hat

### Python version

3.9.12

### CUDA/cuDNN version

CUDA 11.1 / cuDNN 8.2.1.32

### Current Behaviour?

```shell
Creating a model within the scope of a MirroredStrategy results in an OSError during memory cleanup at the end of the program. I would expect it to run without an error.

Others seem to have encountered a similar issue:
- https://stackoverflow.com/questions/66954494/oserror-errno-9-bad-file-descriptor-in-tensorflow-estimater-when-deploying-mo
- https://sm-stackoverflow.azurefd.net/questions/71965552/problem-when-saving-model-after-mirroredstrategy-in-keras

I encountered this error following the example at
https://www.tensorflow.org/guide/distributed_training#use_tfdistributestrategy_with_keras_modelfit.
After getting the error, I trimmed it down to the included minimal example.
```

### Standalone code to reproduce the issue

```shell
'''
Environment setup
>> conda create -n test_tf -c conda-forge python=3.9.12 tensorflow=2.8.0 cudnn=8.2.1.32
>> export CUDA_VISIBLE_DEVICES=""
'''
import os
os.environ['CUDA_VISIBLE_DEVICES'] = "0"

import tensorflow as tf
from tensorflow import keras

def main():
    print('GPUs:', tf.config.list_physical_devices('GPU'))
    with tf.distribute.MirroredStrategy().scope():
        model = tf.keras.Sequential([tf.keras.layers.Dense(1, input_shape=(1,))])
    print('Output shape:', model.output_shape)

if __name__ == '__main__':
    main()
```

### Relevant log output

```shell
GPUs: [PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
2022-05-21 14:16:41.133359: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-05-21 14:16:41.633403: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0
Output shape: (None, 1)
Exception ignored in: <function Pool.__del__ at 0x2aabb1d85670>
Traceback (most recent call last):
  File "~/.conda/envs/test_tf/lib/python3.9/multiprocessing/pool.py", line 268, in __del__
    self._change_notifier.put(None)
  File "~/.conda/envs/test_tf/lib/python3.9/multiprocessing/queues.py", line 378, in put
    self._writer.send_bytes(obj)
  File "~/.conda/envs/test_tf/lib/python3.9/multiprocessing/connection.py", line 205, in send_bytes
    self._send_bytes(m[offset:offset + size])
  File "~/.conda/envs/test_tf/lib/python3.9/multiprocessing/connection.py", line 416, in _send_bytes
    self._send(header + buf)
  File "~/.conda/envs/test_tf/lib/python3.9/multiprocessing/connection.py", line 373, in _send
    n = write(self._handle, buf)
OSError: [Errno 9] Bad file descriptor
```
</details>

**Comment by hustsheng** [comment]
i meet it too, dose it has any solution?

**Comment by sachinprasadhs** [comment]
@alexarmstrongvi , Could you try with the latest nightly version to see the changes from the above linked fix. Thanks!

**Comment by alexarmstrongvi** [comment]
@sachinprasadhs, the issue is fixed when I run on tf_nightly-2.10.0.dev20220616. Thanks!
